### PR TITLE
[BREAKING CHANGE] Enhance genezio analyze env vars

### PR DESCRIPTION
**Pull request by andreia-oca**
# Pull Request Template

## Type of change

-   [ ] 🍕 New feature
-   [ ] 🐛 Bug Fix
-   [ ] 🔥 Breaking change
-   [x] 🧑‍💻 Improvement
-   [ ] 📝 Documentation Update

## Description

This PR makes an improvement on reporting the environment variables to the dashboard.
Now every set of environment variables is mapped to the corresponding component.
The mapping is done using the `path` which must be the same for the component and for the `.env.example` file.

```
export type FrameworkReport = {
    backend?: BaseComponent[];
    frontend?: BaseComponent[];
    ssr?: BaseComponent[];
    services?: FrameworkReportService[];
};

export type BaseComponent = {
    component: string;
    path?: string;
    environment?: ProjectEnvironment[];
};
```

This is a breaking change for the frontend because it updates the objects passed to the dashboard - this PR should be merged with it's companion on the frontend side (wip).

The environment variables are retrieved and set using an OpenAI agent, if that's not available (no OpenAI key found), the fallback is to parse and load each `.env.example` as it is (with keys and potentially default values).

## Checklist

-   [x] My code follows the contributor guidelines of this project;
-   [ ] I have updated the documentation;
-   [ ] I have added tests;
-   [x] New and existing unit tests pass locally with my changes;

### CODE REVIEW

File: src/commands/analyze/command.ts [hunk 1]
```diff
@@ -97,10 +99,23 @@ export enum SUPPORTED_FORMATS {
 export const DEFAULT_FORMAT = SUPPORTED_FORMATS.TEXT;
 export const DEFAULT_CI_FORMAT = SUPPORTED_FORMATS.JSON;
 
-export const KEY_FILES = ["package.json", "requirements.txt", "pyproject.toml"];
 export const KEY_DEPENDENCY_FILES = ["package.json", "requirements.txt", "pyproject.toml"];
 export const ENVIRONMENT_EXAMPLE_FILES = [".env.template", ".env.example", ".env.local.example"];
```
Comment (cristim67) [inline src/commands/analyze/command.ts:103]: can you also add `.env.sample`

File: src/commands/analyze/command.ts [hunk 2]
```diff
@@ -637,6 +710,71 @@ export async function analyzeCommand(options: GenezioAnalyzeOptions) {
...
 
+async function analyzeEnvironmentFilesConcurrently(
+    envExampleFiles: Map<string, string>,
+    genezioConfig: RawYamlProjectConfiguration,
+): Promise<
+    Map<
+        string,
+        {
+            environmentVariables: ProjectEnvironment[];
+        }
```
Comment (cristim67) [inline src/commands/analyze/command.ts:721]: The return type `Map<string, { environmentVariables: ProjectEnvironment[] }> ` is repeated multiple times. Define it as a separate type for maintainability:

```
type EnvironmentAnalysisResult = Map<string, { environmentVariables: ProjectEnvironment[] }>;
```

File: src/commands/analyze/command.ts [hunk 3]
```diff
@@ -637,6 +710,71 @@ export async function analyzeCommand(options: GenezioAnalyzeOptions) {
     log.info(result);
 }
 
+async function analyzeEnvironmentFilesConcurrently(
```
Comment (cristim67) [inline src/commands/analyze/command.ts:713]: Handle this edge case:

```
  if (envExampleFiles.size === 0) {
        return new Map();
    }
    
 ```

**Review by cristim67 — APPROVED** [review]

File: src/commands/analyze/command.ts (hunk 3 above)
Comment (andreia-oca) [reply src/commands/analyze/command.ts:713]: Nice catch <3

**Pull request merged**